agent/plugins/modules/preset_config.py

```diff
@@ -0,0 +1,444 @@
+
+
+def generate_preset_base(home_dir, ci_token, ci_server, working_dir, ssh_pub_key, pull_secret,
+                        openshift_release_stream, openshift_release_type, openshift_version,
+                        cluster_name, base_domain):
+
+    cluster_domain = cluster_name+"."+base_domain
+    ocp_dir = working_dir+"/ocp/"+cluster_name
+    manifests_dir = ocp_dir+"/manifests"
+    provisioning_host_user = getuser()
+    provisioning_network_name = cluster_name+"pr"
+    baremetal_network_name = cluster_name+"bm"
+    local_registry_dns_name = "virthost."+cluster_name+"."+base_domain
+    registry_dir = working_dir+"/registry"
+    registry_creds = home_dir+"/private-mirror-"+cluster_name+".json"
+    mirror_log_file = registry_dir+"/"+cluster_name+"-image_mirror-"+openshift_release_stream+"-"+openshift_release_type+".log"
+
+    facts = {
+        'ci_token': ci_token,
+        'ci_server': ci_server,
+        'working_dir': working_dir,
+        'pull_secret': pull_secret,
+        'ssh_pub_key': ssh_pub_key,
+        'openshift_release_stream': openshift_release_stream,
+        'openshift_release_type': openshift_release_type,
+        'openshift_version': openshift_version,
+        'cluster_name': cluster_name,
+        'base_domain': base_domain,
+        'cluster_domain': cluster_domain,
+        'ocp_dir': ocp_dir,
+        'manifests_dir': manifests_dir,
+        'provisioning_host_user': provisioning_host_user,
+        'provisioning_network_name': provisioning_network_name,
+        'baremetal_network_name': baremetal_network_name,
+        'local_registry_dns_name': local_registry_dns_name,
+        'registry_dir': registry_dir,
+        'registry_creds': registry_creds,
+        'mirror_log_file': mirror_log_file
+    }
+
+    return facts
+
+def determine_cluster_topology(cluster_topology, resource_profile, extra_workers_profile):
+    num_masters, num_workers, num_extra_workers = 0, 0, 0
+    master_memory, master_disk, master_vcpu = None, None, None
+    worker_memory, worker_disk, worker_vcpu = None, None, None
+    extra_worker_memory, extra_worker_disk, extra_worker_vcpu = None, None, None
+    apply_extra_workers = None
+    node_hostname_static_ip = []
+
+    # Define the number of nodes based off topology
+    if cluster_topology == 'ha':
+        num_masters, num_workers, num_extra_workers = 3, 2, 0
+    elif cluster_topology == 'compact':
+        num_masters, num_workers, num_extra_workers = 3, 0 ,0
+    elif cluster_topology == 'sno':
+        num_masters, num_workers, num_extra_workers = 1, 0, 0
+    else:
+        pass
+
+    if extra_workers_profile == 'none':
+        num_extra_workers, apply_extra_workers = 0, None
+    elif extra_workers_profile == 'day2inactive':
+        num_extra_workers, apply_extra_workers = 1, False
+    elif extra_workers_profile == 'day2active':
+        num_extra_workers, apply_extra_workers = 1, True
+
+    # Define the master node resources
+    if num_masters > 0:
```

Comment:
If `num_masters` it's 0 (or 1 in a non-sno config) it should be considered an error

Reply:
To address this comment we actually don't  get in this state. In lines 236-242 I set the value of num_masters based off the cluster_topology. If someone enters a topology not listed - the Ansible module fails because of invalid choice. I think I can add some unit tests that showcase this failure today

Reply:
Ack sounds fine, then probably this check may be redundant then, if we are sure to have a valid `num_master` set when reaching this point?

Reply:
🤔 Yes I suppose this is redundant in its current form.